devito/types/dimension.py

```diff
@@ -892,6 +895,18 @@ def condition(self):
     def indirect(self):
         return self._indirect
 
+    @property
+    def fact_index(self):
+        if self.condition is None or self._factor is None:
+            return self.index
+
+        # This is the corner case where both a condition and a factor are provided
+        # the index will need to be `self.parent - min(self.condition)` to avoid
+        # shifted indexing. E.g if you have `factor=2` and `condition=Ge(time, 10)`
+        # then the lowered index needs to be `(time - 10)/ 2`
+        ltkn = relational_min(self.condition, self.parent)
```

Comment:
ltkn stands for "left thickness" -- is it really the name u wanna use here?

(nitpicking)

Reply:
Yeah that's intentional that the actual left thickness that is required to be shifted by.